src/Umbraco.Core/Persistence/EntityNotFoundException.cs

```diff
@@ -1,38 +1,95 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using System.Runtime.Serialization;
 
 namespace Umbraco.Core.Persistence
 {
-
     // TODO: Would be good to use this exception type anytime we cannot find an entity
 
     /// <summary>
-    /// An exception used to indicate that an umbraco entity could not be found
+    /// An exception used to indicate that an Umbraco entity could not be found.
     /// </summary>
+    /// <seealso cref="System.Exception" />
+    [Obsolete("This exception is only used in MediaController and could be replaced with an HTTP 404 error instead.")]
+    [Serializable]
     public class EntityNotFoundException : Exception
```

Comment:
I've marked `EntityNotFoundException` obsolete, as it's only used once in an API controller and could be replaced with a generic HTTP 404 error:
https://github.com/umbraco/Umbraco-CMS/blob/56545ad7f1fdee3c2b27e725c89378ccb947f65a/src/Umbraco.Web/Editors/MediaController.cs#L818